app/session.server.ts

```diff
@@ -109,3 +109,30 @@ export async function getCurrentUser(session: Session): Promise<User | null> {
 
   return await getUserByUid(claims.uid);
 }
+
+/**
+ * Returns the user that's currently logged in or redirects to the path
+ * provided (defaults to /login).
+ */
+export async function getCurrentUserOrRedirect(
+  request: Request,
+  redirectTo = "/login"
+) {
+  const session = await getSession(request.headers.get("Cookie"));
+  const user = await getCurrentUser(session);
+
+  if (!user) {
+    session.unset("idToken");
+    session.unset("accessToken");
+    throw redirect(redirectTo, {
```

Comment:
By `throw`ing a redirect, we guarantee that a `User` will be returned wherever this code is consumed. This means we never have to write a check like this:

```ts
const user = getCurrentUserOrRedirect(request);
if (user) {
  // ... do something
}
```